### DISCUSSION TITLE
Attribute aliases

### DISCUSSION THREAD

**Discussion by MichelJonkman** [laravel/framework#42488] [Ideas]
A request to add attribute aliases to models. An attribute alias would act just like a normal attribute but with a different name.

```PHP
class User extends \Illuminate\Database\Eloquent\Model
{

    protected $aliases = [
        'identifier' => 'id',
        'username' => 'name'
    ];

}

$model = User::find(1);
echo $model['identifier']; // Print the id attribute of the user
echo $model['username']; // Print the name attribute of the user
```

The implementation would probably have to be in Illuminate\Database\Eloquent\Concerns\HasAttributes and be fairly simple.

Let me know what you think!

Edit: I had switched the key and value around accidentally, fixed that.

**Comment by fgaroby** [comment]
what would be the difference with the `getXXXAttribute()`method ?
```php
public function getIdentifierAttribute(): string
{
    return $this->attributes['id'];
}
```

**Reply by MichelJonkman** [reply]
> what would be the difference with the `getXXXAttribute()`method ?
> 
> ```
> public function getIdentifierAttribute(): string
> {
>     return $this->attributes['id'];
> }
> ```

A shorter syntax and possibly a slight performance improvement. In my use case I have 44 different attributes, writing 88 different methods isn't very practical.

**Comment by SocolaDaiCa** [comment]
@99michel99 in model you can overwrite method __get and set to implemnt Attribute aliases
![image](https://user-images.githubusercontent.com/18243451/170393440-e04f1a02-1ad0-42f0-b9c8-44c3fd705b8e.png)

**Reply by SocolaDaiCa** [reply]
like that
![image](https://user-images.githubusercontent.com/18243451/170394337-636f71dc-d9ef-478c-a792-19c032825b39.png)

**Reply by MichelJonkman** [reply]
Overwriting getAttribute() would be better, however ide-helper doesn't recognize it and I'm afraid other third or first party packages might not be compatible.

**Reply by SocolaDaiCa** [reply]
@99michel99 if you want ide helper can understand it, you need define phpdoc for it inside model

**Reply by MichelJonkman** [reply]
That would be a lot of work to do every time, which is why I'd love it as a feature in Laravel itself
